### ISSUE TITLE
Extras not included in response

### ISSUE THREAD

**Issue by steve-compassdigital** [compassdigital/compassdigital.provider.menu.bamco#32]
After we add an item to the cart with extras, the cart returned doesn't contain any extras.

**Comment by jonathankeebler** [comment]
@steve-compassdigital will post an example to test against

**Comment by steve-compassdigital** [comment]
--> PUT https://api.compassdigital.org/v1/shoppingcart/JJyM1Q8L1vFlQYMlapMjfPmJgXDwRBfKBqkkrjXAse28JoZYJYSGDwNONW88TMePyM2 HTTP/1.1
Content-Type: application/json; charset=UTF-8
Content-Length: 443
{"items":[{"id":"1BkoWK8LM8CRZQ4q6r1jIPNKR8vRzkum0D5ev35kSpaYDryBPkSEErZwrre","options":[{"items":["a2LAdXPQgPi5gaODr4KqivaN4Bq5NQFrZNZZ78zKHW1meZe7eqFQJM7G9DBaiL0KOqEwZv"],"label":{"en":"Sides"}},{"items":["a2LAdXPQgPi5gaODr4KqivaN4Bq5NQFrZNZZ78zKHW1meZe7eqFQJM7G9DBaiL0KOqEwZv"],"label":{"en":"Sides"}},{"items":["D5BmGqQK8Qt9Wp82aKRAcqOZoLLrD2Ce6QyYvjK0T2OPPq106eCoNl62O6RYIRLO2gw"],"label":{"en":"burgers and sandwiches"}}],"quantity":2}]}
--> END PUT (443-byte body)
<-- HTTP/1.1 200  (5574ms)
content-type: application/json; charset=utf-8
content-length: 1039
date: Sat, 09 Dec 2017 23:50:59 GMT
x-amzn-requestid: cb34bf8a-dd3b-11e7-b526-95863a5c8952
x-amzn-remapped-x-amzn-requestid: cb377e46-dd3b-11e7-a982-b369c6621083
access-control-allow-origin: *
x-amzn-remapped-content-length: 1039
x-amzn-remapped-connection: keep-alive
via: 1.1 3572de112011f1b625bb77410b0c5cca.cloudfront.net (CloudFront), 1.1 52e180bc0e5a47e4820e1923633d0ab7.cloudfront.net (CloudFront)
x-amzn-trace-id: sampled=0;root=1-5a2c76de-ce76bfc144249032705375f8
x-amzn-remapped-date: Sat, 09 Dec 2017 23:50:59 GMT
x-cache: Miss from cloudfront
x-amz-cf-id: FR0f-TQgLo7Mzusc_scwp_4ci9liE98GvP3WYnj5bJc_cIGq1DmQKw==
OkHttp-Sent-Millis: 1512863453532
OkHttp-Received-Millis: 1512863459105
{"menu":"BmBO3NG2MGId4PQW7REXsXqdmD4QlQhjdlPk59g9UZqm0Q","brand":"B5XY4Aq2ljsd4PYAQp70HB92M8jO0v","date":{"created":"2017-12-09T23:50:53.720Z","modified":"2017-12-09T23:50:59.278Z"},"id":"JJyM1Q8L1vFlQYMlapMjfPmJgXDwRBfKBqkkrjXAse28JoZYJYSGDwNONW88TMePyM2","items":[{"id":"1BkoWK8LM8CRZQ4q6r1jIPNKR8vRzkum0D5ev35kSpaYDryBPkSEErZwrre","label":{"en":"House-made Black Bean Burger On Whole Grain Bun"},"description":{"en":"Black Bean, Burger, Whole Grain, Fat, Test, Flour, Water, Milk"},"nutrition":{"calories":{"amount":390,"unit":"kcal"}},"certified":{"vegetarian":true},"meta":{"orderable":true,"sub_station":"Burgers And Sandwiches"},"options":[{"label":{"en":"Sides"},"items":[],"min":0,"max":1},{"label":{"en":"Sides"},"items":[],"min":0,"max":1},{"label":{"en":"Sides"},"items":[],"min":0,"max":1},{"label":{"en":"Sides"},"items":[],"min":0,"max":1},{"label":{"en":"burgers and sandwiches"},"items":[],"min":0,"max":1}],"price":{"amount":4.29},"quantity":2}],"sub_total":{"amount":4.29},"taxes":{"amount":0.4},"total":{"amount":4.69}}
<-- END HTTP (1039-byte body)

**Comment by jonathankeebler** [comment]
@steve-compassdigital Those sides of "meatless salami" turned out to be invalid data (removed in https://github.com/compassdigital/compassdigital.provider.menu.bamco/issues/37). I think they were causing the problem here. If you try again with other "options" you find the in the menu item, I think it's going to work for you. If not, please reopen this ticket.

**Comment by steven-tucker** [comment]
We don't have permissions to reopen tickets. No, extras have never worked for us. Did you create a cart which had extras? If so, could you provide the cart id so that we can continue with the UI development? If not, then why was this ticket closed?

**Comment by jonathankeebler** [comment]
Yes, I have created a cart with options. Here is an example curl using the cart you referenced above:
```
curl --request PUT "https://api.compassdigital.org/v1/shoppingcart/JJyM1Q8L1vFlQYMlapMjfPmJgXDwRBfKBqkkrjXAse28JoZYJYSGDwNONW88TMePyM2" \
    --header "Content-Type: application/json" \
    --header "Accept: application/json" \
    --data '{"items":[{"id":"z03ykPpQDpcjK42aNPQAukEmOPLlDwTQ9kX43oXRSZPN0mL2QkSKKDeaDDQ","options":[{"label": {"en": "add salad"}, "items": [{"id": "5wrvjEKLMKfJvAQdLM48IWj4mdowG3CzwXYDPQ3LTJGyyNwQjlCDpjqadql6Fr7QMw1"}],"quantity":1}]}]}'

**Comment by steve-compassdigital** [comment]
I see the swagger has been changed. It was originally defining the options as an array of strings. It has been changed to an array of objects containing a string but no one informed us. Also, your request has a quantity for the group. The quantity is not defined in swagger. Additionally, the quantity is on the group, not the item in the group. I'm assuming we have to pass a value of 1 or more or it will be ignored (required field)?

**Comment by steve-compassdigital** [comment]
Also, please ensure adding a second item of same id with different extras works as expected before closing the ticket.

**Comment by jonathankeebler** [comment]
1. I don't see an old version of the Swagger with options as an array of strings, but glad it's working for you now.
2. Don't worry about the quantity on the group. I think that was just my mistake when I wrote the curl. You don't have to pass it, per the swagger.
3. I have an open ticket for adding second item of same id with different options: https://github.com/compassdigital/compassdigital.service.shoppingcart/issues/30 . Avoid that case for now

Closing this ticket since this particular issue is resolved.
